src/main/java/seedu/address/model/module/ModuleContainsTagsPredicate.java

```diff
@@ -0,0 +1,28 @@
+package seedu.address.model.module;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Module} contains a {@code Tag} matching any of the tag names given.
+ */
+public class ModuleContainsTagsPredicate implements Predicate<Module> {
+    private final List<String> tagNames;
+
+    /**
+     * Creates a new ModuleContainsTagsPredicate to test whether a module contains any tag that
+     * matches any of the tag names in the specified list.
+     *
+     * @param tagNames list of tag names.
+     */
+    public ModuleContainsTagsPredicate(List<String> tagNames) {
+        this.tagNames = tagNames;
+    }
+
+    @Override
+    public boolean test(Module module) {
+        return tagNames.stream().anyMatch(name ->
+              module.getTags().stream().anyMatch(tag ->
+                     tag.tagName.equals(name)));
```

Comment:
Perhaps can improve this by using equalsIgnoreCase? cause as of now if i type in `find foundation`, all the modules with 'Foundation' as tags won't be listed  😢 
```suggestion
                     tag.tagName.equalsIgnoreCase(name)));
```

Reply:
Very very good catch! Thanks and resolved :D